tests/phpunit/Unit/DataValues/DescriptionEncoders/DvDescriptionEncoderFactoryTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace SMW\Tests\DataValues\DescriptionEncoders;
+
+use SMW\DataValues\DescriptionEncoders\DvDescriptionEncoderFactory;
+
+/**
+ * @covers \SMW\DataValues\DescriptionEncoders\DvDescriptionEncoderFactory
+ * @group semantic-mediawiki
+ *
+ * @license GNU GPL v2+
+ * @since 2.3
+ *
+ * @author mwjames
+ */
+class DvDescriptionEncoderFactoryTest extends \PHPUnit_Framework_TestCase {
+
+	protected function tearDown() {
+		DvDescriptionEncoderFactory::getInstance()->clear();
+		parent::tearDown();
+	}
+
+	public function testCanConstruct() {
+
+		$this->assertInstanceOf(
+			'\SMW\DataValues\DescriptionEncoders\DvDescriptionEncoderFactory',
+			new DvDescriptionEncoderFactory()
+		);
+
+		$this->assertInstanceOf(
+			'\SMW\DataValues\DescriptionEncoders\DvDescriptionEncoderFactory',
+			DvDescriptionEncoderFactory::getInstance()
+		);
+	}
+
+	/**
+	 * @dataProvider dataValueProvider
+	 */
+	public function testCanConstructDescriptionEncoder( $dataValue, $expected ) {
+
```

Comment:
I think you are better of just creating a method for each of those assertions. There will some some duplication then, though very non problematic one, and less expensive than the complexity and indirection added here.